### PR TITLE
Fix WKTWriter to emit EMPTY elements in multi-geometries

### LINKED ISSUE #951: WKT round-tripping fail with MULTIPOINT(EMPTY) vs MULTIPOINT EMPTY
A `MULTIPOINT(EMPTY)` geometry has one empty item, while `MULTIPOINT EMPTY` is the same type but no items. They round-trip fine with WKB, but not with WKT. Here is a bash loop to compare the outputs using GEOS 3.12.0:
```bash
for geom_type in MULTIPOINT MULTILINESTRING MULTIPOLYGON; do
  geom="$geom_type(EMPTY)"
  echo "input: $geom"
  echo -n "  wkt: "
  ./bin/geosop -a "$geom" -f wkt
  echo -n "  wkb: "
  ./bin/geosop -a "$geom" -f wkb
done
```
shows the unexpected WKT differences
```
input: MULTIPOINT(EMPTY)
  wkt: MULTIPOINT EMPTY
  wkb: 0104000000010000000101000000000000000000F87F000000000000F87F
input: MULTILINESTRING(EMPTY)
  wkt: MULTILINESTRING EMPTY
  wkb: 010500000001000000010200000000000000
input: MULTIPOLYGON(EMPTY)
  wkt: MULTIPOLYGON EMPTY
  wkb: 010600000001000000010300000000000000
```
Meanwhile, these are all expected:
```bash
for geom_type in MULTIPOINT MULTILINESTRING MULTIPOLYGON; do
  geom="$geom_type EMPTY"
  echo "input: $geom"
  echo -n "  wkt: "
  ./bin/geosop -a "$geom" -f wkt
  echo -n "  wkb: "
  ./bin/geosop -a "$geom" -f wkb
done
```
shows expected zero-length collections:
```
input: MULTIPOINT EMPTY
  wkt: MULTIPOINT EMPTY
  wkb: 010400000000000000
input: MULTILINESTRING EMPTY
  wkt: MULTILINESTRING EMPTY
  wkb: 010500000000000000
input: MULTIPOLYGON EMPTY
  wkt: MULTIPOLYGON EMPTY
  wkb: 010600000000000000
```
There are no issues with GeometryCollection, which can round-trip `GEOMETRYCOLLECTION(POINT EMPTY)` without issue. PostGIS seems to handle these geometries as expected.

I'm almost certain this has been brought up somewhere, but I cannot find this. Empty strikes again!

### PR DESCRIPTION
**Pull request by mwtoews**
Closes #951

### CODE REVIEW

File: tests/unit/io/WKTWriterTest.cpp [hunk 1]
```diff
@@ -405,4 +405,57 @@ void object::test<14>
...
 
+// Test multi-part geometries with zero or more empty parts
+// https://github.com/libgeos/geos/issues/951
+template<>
+template<>
+void object::test<15>
+()
+{
+    // Run all combinations of atomic multi-geometry types and dimensions
+    std::vector<std::string> geom_types{"POINT", "LINESTRING", "POLYGON"};
```
[the remark this answers is not in the document]
Comment (mwtoews) [reply tests/unit/io/WKTWriterTest.cpp]: Sure, done.

**Comment by dr-jts** [comment]
Just to be clear: this is fixing the WKTY parser so that it preserves empty elements in Multi-geometries?

**Comment by mwtoews** [comment]
the WKT parser is fine, just issues with the WKT writer.

**Comment by dr-jts** [comment]
Perhaps a better title is "Fix WKTWriter to emit EMPTY elements in Multi-geometries"?

**Pull request merged**